File: engine/b/src/main/java/net/crazyinvestor/engine_b/repository/SymbolRepository.java
package net.crazyinvestor.engine_b.repository;

import net.crazyinvestor.engine_b.entity.Exchange;
import net.crazyinvestor.engine_b.entity.Symbol;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface SymbolRepository extends JpaRepository<Symbol, Long> {
    Optional<Symbol> findByExchangeAndName(Exchange exchange, String name);
}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/deprecated/repository/SymbolRepository.kt
package net.crazyinvestor.ticker_producer_engine.deprecated.repository

import net.crazyinvestor.ticker_producer_engine.deprecated.entity.Exchange
import net.crazyinvestor.ticker_producer_engine.deprecated.entity.Symbol
import org.springframework.stereotype.Repository
import java.util.*

@Repository
class SymbolRepository {
    fun findByExchangeAndName(exchange: Exchange, name: String): Optional<Symbol> {
        println("MOCK REPOSITORY3 $exchange $name")
        return Optional.of(Symbol(null, name, exchange, null))
    }

    fun save(symbol: Symbol): Symbol {
        println("save() MOCK REPOSITORY3 $symbol")
        return symbol
    }
}